### PR TITLE
Extend commands options master

### PR DESCRIPTION
**Pull request by MosheSheena**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -57,3 +57,4 @@ docs/_build/
 # PyBuilder
 target/
 python-rrmngmnt.iml
+/.venv/
```
Comment (myakove) [inline .gitignore:60]: How this is related to this PR?

File: .travis.yml [hunk 2]
```diff
@@ -3,8 +3,8 @@ sudo: required
 services:
   - docker
 python:
-- '3.5'
 - '3.6'
+- '3.7'
```
Comment (myakove) [inline .travis.yml:7]: ditto

File: rrmngmnt/nmcli.py [hunk 3]
```diff
@@ -237,6 +221,7 @@ def add_ethernet_connection(
         ipv6_method=None,
         ipv6_addr=None,
         ipv6_gw=None,
+        **kwargs
```
Comment (myakove) [inline rrmngmnt/nmcli.py:224]: kwargs hide a lot of things, any reason to use it and not specify the args that you need here?

File: rrmngmnt/nmcli.py [hunk 4]
```diff
@@ -656,23 +647,33 @@ def _get_str_value(value):
...
 
         return common_options
 
+    @staticmethod
+    def _type_options_builder(**kwargs):
+        """
+        Builds nmcli command type options.
+
+        Args:
+            *args: positional arguments.
```
Comment (myakove) [inline rrmngmnt/nmcli.py:669]: where do you get args here?

File: rrmngmnt/nmcli.py [hunk 5]
```diff
@@ -656,23 +647,33 @@ def _get_str_value(value):
...
+            **kwargs: keyword arguments.
```
Comment (myakove) [inline rrmngmnt/nmcli.py:670]: **kwargs: keyword arguments. is not the way to write a docstring.
please specify what are the kwargs expected here.

File: setup.cfg [hunk 6]
```diff
@@ -14,8 +14,8 @@ classifier =
     Operating System :: POSIX
     Programming Language :: Python
     Programming Language :: Python :: 3
-    Programming Language :: Python :: 3.5
     Programming Language :: Python :: 3.6
+    Programming Language :: Python :: 3.7
```
Comment (myakove) [inline setup.cfg:18]: not related to this PR

File: tox.ini [hunk 7]
```diff
@@ -1,8 +1,8 @@
 [tox]
-envlist = py35,py36,pep8
+envlist = py36,py37,pep8
```
Comment (myakove) [inline tox.ini:2]: ditto

File: tox.ini [hunk 8]
```diff
@@ -1,8 +1,8 @@
...
 [tox:travis]
-3.5 = py35, pep8
 3.6 = py36, pep8
+3.7 = py37, pep8
```
Comment (myakove) [inline tox.ini:5]: ditto

**Review by myakove — CHANGES_REQUESTED** [review]
This PR should be split into 3 PRs.
1, python version chnges.
2. Format strings.
3. introduce _type_options_builder

**Comment by MosheSheena** [comment]
> This PR should be split into 3 PRs.
> 1, python version chnges.
> 2. Format strings.
> 3. introduce _type_options_builder

I thought I could split it into several commits instead of multiple PRs, but I'm fine with splitting it.
How can I preserve all your comments if I drop this PR and start over?

**Pull request closed without merging**